src/module.c

```diff
@@ -728,39 +786,98 @@ int TSDB_createRule(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
             ctx, "TSDB: the source key and destination key should be different");
     }
 
-    // First we verify the source is not a destination
+    // 1. Verify the source is not a destination
     Series *srcSeries;
     RedisModuleKey *srcKey;
-    const int statusS =
-        GetSeries(ctx, srcKeyName, &srcKey, &srcSeries, REDISMODULE_READ | REDISMODULE_WRITE);
+    const int statusS = GetSeries(ctx,
+                                  NULL,
+                                  srcKeyName,
+                                  SERIES_RELATION_NO_RELATION,
+                                  &srcKey,
+                                  &srcSeries,
+                                  REDISMODULE_READ | REDISMODULE_WRITE,
+                                  false);
     if (!statusS) {
         return REDISMODULE_ERR;
     }
     if (srcSeries->srcKey) {
-        return RTS_ReplyGeneralError(ctx, "TSDB: the source key already has a source rule");
+        // still src series might be deleted, check it
+        Series *_srcSeries;
+        RedisModuleKey *_srcKey;
+        const int status = GetSeries(ctx,
```

Comment:
@OfirMos another case as I mentioned in ts.info, this should be handled inside GetSeries in line 792